core/friends.js

```diff
@@ -154,13 +154,27 @@ exports.removeFriend = function(id, cb) {
 };
 
 // Grabs what names it can from bot.users and applies them to the friends list.
-exports.updateFriendsNames = function() {
-  for (var friend in friends) {
-    if (bot.users.hasOwnProperty(friend) && friends.hasOwnProperty(friend)) {
-      friends[friend].name = bot.users[friend].playerName;
-    }
+/*
+* bot.users was old, new thing is bot.friends
+* bot.friends is an object of the user's steamId and the friend type, ie: '765...': 3,
+*/
+exports.updateFriendsNames = function(steamId, username) {
+  console.log(bot.friends);
+  // console.log(bot.requestFriendData(bot.friends));
+  if (steamId !== null) {
+    exports.set(steamId,'name', username);
   }
-  exports.save();
+  else {
+    exports.addFriend(steamId);
```

Comment:
Would it be null if called from one of the places where the parameters are omitted?

I was trying a different solution:

`+botFriends.on('personaState', function(resp) {`
`console.log(resp.friendid + ', ' + resp.player_name);`
 `friends.updateFriendsNames(resp.friendid, resp.player_name);`
  `});`

Calling from here, I think it would update a friend's name or add the user to the friend's list if they were a newly encountered user.  No specific reason, just a possibility.